InMobi/src/main/java/com/mopub/nativeads/InMobiNative.kt

```diff
@@ -0,0 +1,334 @@
+
+        InMobiAdapterConfiguration.initializeInMobi(
+            serverExtras,
+            context,
+            object : InMobiAdapterConfiguration.InMobiInitCompletionListener {
+                override fun onSuccess() {
+                    loadNative(serverExtras, customEventNativeListener)
+                }
+
+                override fun onFailure(error: Throwable) {
+                    val msg = "InMobi interstitial request failed due to InMobi initialization failed with an exception."
+                    onError(customEventNativeListener, msg)
+                }
+            })
+    }
+
+    private fun onError(customEventNativeListener: CustomEventNativeListener, errorMsg: String) {
+        customEventNativeListener.onNativeAdFailed(NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR)
+        MoPubLog.log(
+            getAdNetworkId(),
+            AdapterLogEvent.LOAD_FAILED,
+            TAG,
+            NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR,
+            errorMsg
+        )
+    }
+
+    private fun getAdNetworkId(): String {
+        return mPlacementId.toString() ?: ""
+    }
+
+    private fun loadNative(serverExtras: Map<String, String>, customEventNativeListener: CustomEventNativeListener) {
+//        InMobiSdk.updateGDPRConsent(InMobiGDPR.getGDPRConsentDictionary())
+        InMobiSdk.setLogLevel(InMobiSdk.LogLevel.DEBUG)
+        inMobiStaticNativeAd = try {
+            val context = mContext?.get()
+            if (context == null) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "Context passed to the Adapter is null or might have garbage collected")
+                customEventNativeListener.onNativeAdFailed(NativeErrorCode.UNSPECIFIED)
+                return
+            }
+            InMobiNativeAd(context, customEventNativeListener, mPlacementId)
+        } catch (e: SdkNotInitializedException) {
+            MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "Error while creating InMobiNative Object. " + e.message)
+            handleNativeInitializationFailure()
+            return
+        }
+        inMobiStaticNativeAd?.setExtras(InMobiAdapterConfiguration.inMobiTPExtras)
+
+        val adMarkup = serverExtras[DataKeys.ADM_KEY]
+        if (adMarkup != null) {
+            MoPubLog.log(
+                AdapterLogEvent.CUSTOM, InMobiBanner.ADAPTER_NAME,
+                "Ad markup for InMobi banner ad request is present. Will make Advanced Bidding ad request " +
+                        "using markup: " + adMarkup
+            )
+            inMobiStaticNativeAd?.loadAd(adMarkup?.toByteArray())
+        } else {
+            MoPubLog.log(
+                AdapterLogEvent.CUSTOM, InMobiBanner.ADAPTER_NAME,
+                "Ad markup for InMobi banner ad request is not present. Will make traditional ad request "
+            )
+            inMobiStaticNativeAd?.loadAd()
+        }
+
+        inMobiStaticNativeAd?.let { STATIC_MAP.putIfAbsent(inMobiStaticNativeAd.hashCode(), it) }
```

Comment:
I don't see any use case for putting the ad instance in a static map. This is a legacy code but I can't find the reason for adding this map. Can you also check and if it's not needed let's remove this and test once.

Reply:
I also can't see any use of STATIC_MAP, it is only storing instances of InMobiNativeAd class

Is it done to keep a reference to each Ad instance so that GC does not collect them?
Remove this for now